src/app/sidenav/sidenav.component.html

```diff
@@ -3,9 +3,20 @@
     <mat-nav-list>
       <a mat-list-item [routerLink]="['/']">Settings</a>
       <mat-divider></mat-divider>
-      <a mat-list-item [routerLink]="['/']">Community 1</a>
-      <a mat-list-item [routerLink]="['/']">Community 2</a>
-      <a mat-list-item [routerLink]="['/']">Community 3</a>
+      <mat-list-item *ngFor="let communityFav of communitiesFavorited; let j = index">
+        <a mat-line *ngIf="communityFav.isFavorited" [routerLink]="['/']">
+           {{ communityFav.communityname }}
+        </a>
+        <button mat-icon-button (click)="removeFavoriteFromFav(communityFav)" ><mat-icon class="favorited">star</mat-icon></button>
+      </mat-list-item>
+      <mat-divider></mat-divider>
+      <mat-list-item *ngFor="let community of communities">
+        <a mat-line [routerLink]="['/']">
+           {{ community.communityname }}
+        </a>
+        <button mat-icon-button (click)="favorite(community)" *ngIf="!community.isFavorited"><mat-icon>star_border</mat-icon></button>
```

Comment:
You can merge these two buttons together by having a conditional style. You are flipping the favourite status anyways, so combining the functions should not be very difficult and you can name it `toggleFavourite`

```html
<button mat-icon-button (click)="toggleFavourite(community)"><mat-icon [style.color]="community.isFavourited() ? '#FFD700': ''">star_border</mat-icon></button>
```